code/Tree.cs

```diff
@@ -37,15 +38,90 @@ public class Tree
         }
 
         /// <summary>
+        /// Gives the number of children of a vertex.
+        /// </summary>
+        /// <param name="vertexId"> The id of the vertex parent. </param>
+        /// <returns> Returns the number of children vertices (0 if there are none).
+        /// If the vertex parent does not exist, it returns 0 .</returns>
+        public int NbChildren(int vertexId)
+        {
+            int nbChildren = 0;
+
+            if (!children.ContainsKey(vertexId))
+                nbChildren = 0;
+            else
+            {
+                if (children[vertexId] != null)
+                    nbChildren = children[vertexId].Count();
+            }
+
+            return nbChildren;
+        }
+
+        /// <summary>
+        /// Gives the parent of the child in the parameter.
+        /// </summary>
+        /// <param name="vertexId"> The identifier of the child. </param>
+        /// <returns> Returns the identifier of the parent.
+        /// In case the parameter doesn't exist, it raises an exception and returns null. </returns>
+        public int? Parent(int vertexId)
+        {
+            int? parentId;
+
+            try
+            {
+                parentId = parent[vertexId];
+            }catch(KeyNotFoundException)
+            {
+                parentId = null;
+            }
+            
+            return parentId;
+        }
+
+        /// <summary>
+        /// Gives a list of the specified id's Children.
+        /// </summary>
+        /// <param name="vertexId"> The identifier of the parent. </param>
+        /// <returns> Returns a list of the parent's children.
+        /// If the identifier doesn't have children, it returns an empty list.
+        /// If the identifier doesn't exist, it returns null. </returns>
+        public List<int> Children(int vertexId)
+        {
+            List<int> listOfChildren;
+
+            if (!parent.ContainsKey(vertexId))
+                listOfChildren = null;
+            else
+            {
+                if (children[vertexId] != null)
+                    listOfChildren = children[vertexId];
+                else
+                    listOfChildren = new List<int>() { };
+            }
+            return listOfChildren;
+        }
+
+        /// <summary>
         /// Counts the number of elements in the tree.
         /// </summary>
         /// <returns>Number of vertices</returns>
         public int Count() {
-            int count = children.Count();
+            int count = parent.Count();
             return count;
         }
 
         /// <summary>
+        /// Counts the number of vertices in the tree.
+        /// </summary>
+        /// <returns> Returns the total number of vertices. </returns>
+        public int NbVertices()
+        {
+            int nbVertices = parent.Count();
```

Comment:
@pradal : I wrote in my side notes that it was a duplicate of the function Count() and I wanted to ask you about it indeed.